### ISSUE TITLE
[java] UnusedImports with static imports on subclasses

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#3132]
**Affects PMD Version:** 
6.31.0

**Rule:**

[UnusedImports](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedimports)

**Description:**

 I have a static import of all methods in a class but the methods referenced in the code are from its base class. The rule flags the derived class' import as unused despite the compiler recognising the import as valid.

This is unexpected.

Note:

In my opinion, this is fine since it's (strictly) true that the methods do not belong to the specified import. I interpret it as forcing the user the use the correct import. If there are methods available in the derived class and if even one is referenced, then it is used and there is no complation or rule violation for the base class' methods. It's a corner case that probably will occur only in error.
I think the rule must recognise that the parent class' methods are being referenced instead and suggest that the parent class be used instead. It's more of a case of MisusedImport rather than UnusedImport.

**Code Sample demonstrating the issue:**

```java
class Hello {
    public static void sayHello() {
        //...
    }
}

class HelloMore extends Hello {
    // ...
}

import static HelloMore.*; // flagged as unused

public class Test {
    public static void main(String... args) {
        sayHello();
    }
}
```

**Expected outcome:**
No violation should be reported. 

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Referred from: https://github.com/pmd/pmd/discussions/3130#discussioncomment-394030
